paddle/fluid/eager/grad_node_info.h

```diff
@@ -253,14 +254,14 @@ class GradNodeBase {
    * **/
   inline bool GradientHooksRegistered() { return !gradient_hooks_.empty(); }
 
-  std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>
+  std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>&
   GetGradientHookFuntions() {
     VLOG(6) << "GetGradientHookFuntions ";
     return gradient_hooks_;
   }
 
   void SetGradientHookFuntions(
-      std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>
+      std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>&
```

Comment:
In my opinion, GetGradientHookFuntions  and  SetGradientHookFuntions  can use reference type  for passing value to reduce copy time.